### PR TITLE
Hopefully fixes the bug where a traitor AI fails a "Nothing" objective

### PR DESCRIPTION
**Pull request by harmonyn**
Fixes https://github.com/tgstation/tgstation/issues/22967.
Fixes https://github.com/tgstation/tgstation/issues/18966.

What is happening there, if I'm right:

- The traitor AI rolls the "yandere" objective set (protect someone and maroon them). I tested this part, it's always this one that gives the Nothing (and the only way the AI can get a protect objective anyway), and the Nothing is in fact a maroon objective in VV
- The protect objective is generated normally, but the maroon objective has its target set manually (to the same person) instead of via find_target().
- update_explanation_text() gets called in find_target, which means it never happens for the Maroon objective and the text remains "Nothing" (the default). This is why it doesn't say Free Objective, which is what happens when it can't find a target
- The maroon objective works normally, but since the AI doesn't know it's supposed to be marooning someone, it probably doesn't try to maroon them, so they fail it. Maroon is set to succeed by default, this is the only way it *can* fail

haven't tested this but I'm pretty sure this is what's going on.

### CODE REVIEW

File: code/game/gamemodes/traitor/traitor.dm [hunk 1]
```diff
@@ -115,6 +115,7 @@
 					var/datum/objective/maroon/yandere_two = new
 					yandere_two.owner = traitor
 					yandere_two.target = yandere_one.target
+					yandere_two.update_explanation_text() // normally called in find.target()
```
Comment (RemieRichards) [inline code/game/gamemodes/traitor/traitor.dm]: find_target.

**Comment by RemieRichards** [comment]
Your deduction seems correct.
This is a case where I wish we had properties, then any setting of ```target``` could update the text, curse DM.

File: code/game/gamemodes/traitor/traitor.dm (hunk 1 above)
Comment (bgobandit) [inline code/game/gamemodes/traitor/traitor.dm]: it's a comment ¯\_(ツ)_/¯

**Comment by KorPhaeron** [comment]
Thanks for fixing this

**Pull request merged**